File: src/main/java/net/frozenorb/potpvp/util/serialization/LocationSerializer.java
package net.frozenorb.potpvp.util.serialization;

import com.mongodb.BasicDBObject;
import net.frozenorb.potpvp.PotPvPRP;
import org.bukkit.Location;
import org.bukkit.World;

public final class LocationSerializer {

    // Static utility class -- cannot be created.
    private LocationSerializer() {
    }

    public static BasicDBObject serialize(Location location) {
        if (location == null) {
            return (new BasicDBObject());
        }

        BasicDBObject dbObject = new BasicDBObject();

        dbObject.put("world", location.getWorld().getName());
        dbObject.put("x", location.getX());
        dbObject.put("y", location.getY());
        dbObject.put("z", location.getZ());
        dbObject.append("yaw", location.getYaw());
        dbObject.append("pitch", location.getPitch());

        return (dbObject);
    }

    public static Location deserialize(BasicDBObject dbObject) {
        if (dbObject == null || dbObject.isEmpty()) {
            return (null);
        }

        World world = PotPvPRP.getInstance().getServer().getWorld(dbObject.getString("world"));
        double x = dbObject.getDouble("x");
        double y = dbObject.getDouble("y");
        double z = dbObject.getDouble("z");
        int yaw = dbObject.getInt("yaw");
        int pitch = dbObject.getInt("pitch");

        return (new Location(world, x, y, z, yaw, pitch));
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/util/serialization/LocationSerializer.kt
package net.frozenorb.potpvp.kt.util.serialization

import com.mongodb.BasicDBObject
import net.frozenorb.potpvp.PotPvPRP
import org.bukkit.Location

object LocationSerializer {

    @JvmStatic
    fun serialize(location: Location?): BasicDBObject {
        if (location == null) {
            return BasicDBObject()
        }

        val dbObject = BasicDBObject()
        dbObject["world"] = location.world.name
        dbObject["x"] = location.x
        dbObject["y"] = location.y
        dbObject["z"] = location.z
        dbObject.append("yaw", location.yaw)
        dbObject.append("pitch", location.pitch)
        return dbObject
    }

    @JvmStatic
    fun deserialize(dbObject: BasicDBObject?): Location? {
        if (dbObject == null || dbObject.isEmpty()) {
            return null
        }

        val world = PotPvPRP.getInstance().server.getWorld(dbObject.getString("world"))
        val x = dbObject.getDouble("x")
        val y = dbObject.getDouble("y")
        val z = dbObject.getDouble("z")
        val yaw = dbObject.getInt("yaw")
        val pitch = dbObject.getInt("pitch")
        return Location(world, x, y, z, yaw.toFloat(), pitch.toFloat())
    }

}